src/vcpkg-test/configparser.cpp

```diff
@@ -333,15 +418,22 @@ TEST_CASE ("BinaryConfigParser args", "[binaryconfigparser]")
     {
         auto parsed = create_binary_providers_from_configs_pure("files," ABSOLUTE_PATH,
                                                                 std::vector<std::string>{"clear", "clear"});
-        REQUIRE(parsed.has_value());
+        auto state = parsed.value_or_exit(VCPKG_LINE_INFO);
+
+        REQUIRE(state.binary_cache_providers == std::set<std::string>{"files"});
     }
 }
 
 TEST_CASE ("BinaryConfigParser azblob provider", "[binaryconfigparser]")
 {
+    UrlTemplate url_temp;
     {
         auto parsed = create_binary_providers_from_configs_pure("x-azblob,https://azure/container,sas", {});
-        REQUIRE(parsed.has_value());
+        auto state = parsed.value_or_exit(VCPKG_LINE_INFO);
+
+        REQUIRE(state.binary_cache_providers == std::set<std::string>{"azblob"});
+        REQUIRE(url_temp.get_url(state) == std::string{"https://azure/container/{sha}.zip?sas"});
```

Comment:
This should be asserting that this URI is the only one in the read set, and that the write set is empty.